src/runtime/recipe/type-checker.ts

```diff
@@ -150,9 +156,23 @@ export class TypeChecker {
   }
 
   static _tryMergeConstraints(handleType: Type, {type, relaxed, direction}: TypeListInfo, options: {typeErrors?: string[]} = {}): boolean {
-    let [primitiveHandleType, primitiveConnectionType] = Type.unwrapPair(handleType.resolvedType(), type.resolvedType());
+    const [primitiveHandleType, primitiveConnectionType] = Type.unwrapPair(handleType.resolvedType(), type.resolvedType());
```

Comment:
We don't need to unwrapPair if we can unwrapMulti - move this below the handling of handleInnerTypes != null?

Reply:
I presume you are after a clarity of code so that it's organized:
1) If we have multiple types in both arguments, call pairwise.
2) Else, do the work.
Done that, though I needed to make unwrapPair be the first stage of unwrapMulti, which I presume you would be ok with.

This is needed so that if we call _tryMergeConstraints with for example:
handleType: [(~a, ~b)]
connectionType: [(Product {}, Location {})]
We need to first unwrap the collection, and then unwrap the tuple, then merge the constraints of type variables.